### ISSUE TITLE
valid_docs rule failing on `throws` inside a closure in a function.

### ISSUE THREAD

**Issue by nlutsenko** [realm/SwiftLint#451]
The rule is failing on a given function declaration:
```swift
public func performThings(things: (Void throws -> Result)) -> Result
```

From the code - it looks like we are simply doing a regex capture for the `throws` signature, which is technically not correct, as the documentation should be purely on the `throws` that preceeds the function return parameters.

According to the grammar, where optional pieces are italic - it captures the one inside parameter-clauses, but should only look at the one after all of the parameter clauses.

function-signature → parameter-clauses _throws_ ­­_function-result­_

**Comment by jpsim** [comment]
yup, definitely a bug. thanks for reporting!

**Comment by jpsim** [comment]
Fixed in #464
